samigo/samigo-cp/src/java/org/sakaiproject/tool/assessment/contentpackaging/ImportService.java

```diff
@@ -90,27 +90,27 @@ public String unzipImportFile(String filename) {
 	    	entry = (ZipEntry) zipStream.getNextEntry();
 	    	// Get the name of the imported zip file name. The value of "filename" has timestamp append to it.
 	    	String tmpName = filename.substring(filename.lastIndexOf("/") + 1);
-	    	qtiFilename = "exportAssessment.xml";
-	    	ArrayList xmlFilenames = new ArrayList();
+	    	this.setQtiFilename("exportAssessment.xml");
 	    	while (entry != null) {
-	    		String zipName = entry.getName();
-	    		int ix = zipName.lastIndexOf('/');
+	    		String entryName = entry.getName();
+	    		String entryNameTrimmed = entryName.trim();
+	    		int ix = entryName.lastIndexOf('/');
 	    		if (ix > 0) {
-	    			String dirName = zipName.substring(0, ix);
+	    			String dirName = entryName.substring(0, ix);
 	    			if (!dirsMade.contains(dirName)) {
 	    				File d = new File(dir.getPath() + "/" + dirName);
 	    				// If it already exists as a dir, don't do anything
 	    				if (!(d.exists() && d.isDirectory())) {
 	    					// Try to create the directory, warn if it fails
 	    					if (!d.mkdirs()) {
-	    						log.error("unable to mkdir " + dir.getPath() + "/" + dirName);
+	    						log.error("Unable to mkdir " + dir.getPath() + "/" + dirName);
```

Comment:
I'd move dir.getPath() + "/" + dirName into a variable called directory. It's used up above and you'll then just be able to write:
log.error("Unable to mkdir {}", directory);